semantic_steve/js/src/thing/factory.ts

```
@@ -0,0 +1,102 @@
+
+  /**
+   * Creates a Thing object based on the provided input string.
+   * @param thing The name of the thing to create (e.g., "wood_planks", "plains").
+   * @returns {Thing} The created Thing object.
+   * @throws {InvalidThingError} If the input string does not correspond to a valid (supported) thing.
+   */
+  public createThing(thing: string, preferred?: (new(...args: any[]) => Thing)): Thing {
+    if (preferred != null) {
+      switch (preferred.name) {
+        case "Block":
+          const block = this.createBlock(thing);
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)